website/src/controllers/groups.js

```diff
@@ -137,7 +137,25 @@ api.get = function(req, res, next) {
   q.exec(function(err, group){
     if (err) return next(err);
     if (!group && gid!=='party') return res.json(404,{err: "Group not found or you don't have access."});
-    res.json(group);
+
+    //Since we have a limit on how many members are populate to the group, we want to make sure the user is always in the group
+    var userInGroup = _.find(group.members, function(member){ return member._id == user._id; });
+
+    //If the group is private or the group is a party, then the user must be a member of the group based on access restrictions above
+    if (group.privacy === 'private' || gid === 'party') {
+      //If the user is not in the group query, add them
+      if (userInGroup === undefined) { group.members.push(user); }
```

Comment:
I don't think so. The reason I used undefined is because that is the status of the object if `_.find()` doesn't find anything. 

However, this seems to be a bad habit of mine. Is there anywhere I can read documentation on the differences between using !object and other null comparators?